src/coreclr/debug/daccess/daccess.cpp

```diff
@@ -7146,8 +7121,47 @@ CLRDataCreateInstance(REFIID iid,
 #ifdef LOGGING
     InitializeLogging();
 #endif
-    hr = pClrDataAccess->QueryInterface(iid, iface);
 
+    // TODO: [cdac] Remove when cDAC deploys with SOS - https://github.com/dotnet/runtime/issues/108720
+    NonVMComHolder<IUnknown> cdacInterface = nullptr;
+#ifdef CAN_USE_CDAC
+    CLRConfigNoCache enable = CLRConfigNoCache::Get("ENABLE_CDAC");
+    if (enable.IsSet())
+    {
+        DWORD val;
+        if (enable.TryAsInteger(10, val) && val == 1)
+        {
+            // TODO: [cdac] TryGetSymbol is only implemented for Linux, OSX, and Windows.
+            uint64_t contractDescriptorAddr = 0;
+            if (TryGetSymbol(pClrDataAccess->m_pTarget, pClrDataAccess->m_globalBase, "DotNetRuntimeContractDescriptor", &contractDescriptorAddr))
+            {
+                IUnknown* thisImpl;
+                HRESULT qiRes = pClrDataAccess->QueryInterface(IID_IUnknown, (void**)&thisImpl);
+                _ASSERTE(SUCCEEDED(qiRes));
+                CDAC& cdac = pClrDataAccess->m_cdac;
+                cdac = CDAC::Create(contractDescriptorAddr, pClrDataAccess->m_pTarget, thisImpl);
+                if (cdac.IsValid())
+                {
+                    // Get SOS interfaces from the cDAC if available.
+                    cdac.GetSosInterface(&cdacInterface);
+                }
+
+                // Release the AddRef from the QI.
+                pClrDataAccess->Release();
+            }
+        }
+    }
+#endif
+    if (cdacInterface != nullptr)
+    {
+        hr = cdacInterface->QueryInterface(iid, iface);
+    }
+    else
+    {
+        hr = pClrDataAccess->QueryInterface(iid, iface);
+    }
+
+    // Lifetime is now managed by caller (either directly or via cDAC which holds on to the CLRDataAccess instance)
     pClrDataAccess->Release();
```

Comment:
I'm confused about this Release.  If cdacInterface is non-null what is the matching addref that this is releasing?